price-estimator/src/orderbook.rs

```diff
@@ -98,8 +96,13 @@ impl Orderbook {
 
     /// Update the infallible price source with the averaged prices of the external price sources
     /// and the pricegraph prices.
-    async fn update_infallible_price_source(&self, pricegraph: &Pricegraph) -> Result<()> {
-        self.infallible_price_source.update(pricegraph).await
+    async fn update_infallible_price_source(&self, pricegraph: &Pricegraph) {
```

Comment:
Would this method make more sense on the UpdatingInfalliblePriceSource or do you foresee a reason why one would like to update one or the other? Also, should we join the two futures?

Reply:
Maybe this could be `UpdatingInfalliblePriceSource::update(&self, pricegraph: &Pricegraph)`? That way `update_tokens` and `update_prices` could remain private (unless of course, as @fleupold pointed out they will eventually be called separately).

Reply:
I was thinking we might update this in different intervals and they perform different actions to which we might react differently like the different error being logged. I don't think it's important either way but I like this slightly more. (Added the join).